packages/@glimmer/validator/lib/cell.ts

```diff
@@ -0,0 +1,54 @@
+import type { UpdatableTag } from '@glimmer/interfaces';
+
+import { consumeTag } from './tracking';
+import { createUpdatableTag, DIRTY_TAG } from './validators';
+
+type Equality<T> = (a: T, b: T) => boolean;
+
+export class Cell<Value> {
+  static create<T>(value: T, equals?: Equality<T>): Cell<T> {
+    return new Cell(value, equals);
+  }
+
+  #value: Value;
+  readonly #equals?: Equality<Value> | undefined;
+  readonly #tag: UpdatableTag;
+
+  private constructor(value: Value, equals?: Equality<Value>) {
+    this.#value = value;
+    this.#equals = equals;
+    this.#tag = createUpdatableTag();
+  }
+
+  get current() {
+    consumeTag(this.#tag);
+
+    return this.#value;
+  }
+
+  read(): Value {
+    consumeTag(this.#tag);
+
+    return this.#value;
+  }
+
+  set(value: Value): boolean {
```

Comment:
Would it be worth making the set a bounded func? For example:
```ts
  set = (value: Value) =>  boolean {
```

This way we could pass this into templates and not need to create a func just to set the value of a cell.

Reply:
how do you mean?

it's a bug in the vm that this doesn't work:
```hbs
<button {{on 'click' (fn cell.set 'hi')}}></button>
```